### ISSUE TITLE
Configurable Item is still being added to the order even with extension running

### ISSUE THREAD

**Issue by philortiz227** [organicinternet/magento-configurable-simple#121]
Magento version. 1.4.1.1
OrganicInternet SimpleConfigurable Products extension version: 0.7.4.

Issue:

I'm experiencing a very rare glitch where the CONFIGURABLE item still gets added to the cart even if this extension is already installed.  

It happens very rarely. I know it is happening because the SKU of the configurable item will appear in the order instead of the SKU of the simple item. So for some reason the configurable item itself is getting added to a customer's cart instead of the simple item. The price of the item is also shown as the price of the Configurable item, too. 

My theory is that Magento is somehow not running the OrganicInternet extension, or bypassing it for some technical reason, so it ends up reverting to its original behavior of adding configurable items to the shopping cart.

A lot of the orders (in fact, a big majority of them) don't have this problem, but it's quite frustrating to be worried that it will happen once in a blue moon. So that means we still have to double check in case the prices/sku are wrong. 

I think I just need help on how to possibly trace this in the code. I can program in PHP and have done a lot of customization for Magento already, but what I have a hard time with is tracing this particular error. I think this extension is directly related to the error; I just don't know where to look to even guess at what may be causing this problem.

**Comment by jdserrano** [comment]
A client of ours is experiencing the same issue. It is impossible to predict when it will happen because if I do a test order, the issue doesn`t happen.
Magento version: 1.7.0.0
OrganicInternet SimpleConfigurable Products extension version: 0.8.2

On the attached image, Examine sku TEA03014_100

Although this is the sku of the simple product, the price of the configurable product is being used and there is this `Size 100g` which is not in the other items that are using the correct pricing.

Please see the screenshot.
![config_product_price_applied](https://f.cloud.github.com/assets/3135464/32987/e32a327e-5058-11e2-991e-bd570ab9d551.PNG)

**Comment by vishveskrish** [comment]
I am facing same issue !!!

Any one get solution ???

**Comment by aerostah** [comment]
Hello,

I've observed this issue consistently happens with 64-bit Chrome.  It's specific to the add to cart action, you will notice a slight variation of the line item name, where the option text is not appended to the product name on the cart page. The configurable product is being added, not the simple product.. This issue has becoming more prevalent as 64-bit Chrome is no longer in beta and customers use it more often.

This issue is JS related, I haven't debugged yet.  If anyone figured it out, please share your findings.

Thank you,

**Comment by philortiz227** [comment]
I don’t think it’s browser related. For now we programmed a separate class for adding items to the cart, and in this class we use the ‘addproduct’ function, making sure to fill out the required data completely:

$configArr["qty"] = $data->quantity;
$configArr["product"] = $data->pid;
$configArr["cpid"] = $data->parentid;
$configArr["super_attribute"] = $superAttributeArray;

$cart->addProduct($simpleProduct,$configArr);
$cart->save();

configArr is an array, which contains specific data in it:

quantity
data->pid which is the SIMPLE product id
cpid = is the configurable product ID
super_attribute – is a pair of keys/values which refers to the simple product’s association to the configurable parent. You can see these values if you inspect element on your browser, on the drop down that appears for a configurable product options, notice that there are keys/values associated to each option that the customer selects. These key/value pairs are what go to ‘superAttributeArray’. You’ll have to research on how to find the structure of this super_attribute array.

Basically in our end we just use this separate class which completely fills out the needed info when adding a configurable product to the cart. If you use the Magento defaults, there is ‘something’ broken in the OrganicInternet code somewhere which (I think) does not fill out all these values completely. So the configurable product ends up going to your cart as a configurable product only. If you use ‘cart->addproduct’ and fill out the fields I mentioned above, you’ll get consistent results every time. 

I hope it helps. I was dealing with this problem for a few years and took a while to figure out.

Regards,
Philip

**Comment by aerostah** [comment]
Hello,

Phil is correct on it not being related to Chrome 64-bit. We dug some more and found there was an extension conflict.  Mage-World's Free Gift Pro, had xml which outputted to product.info.options.wrapper, product.info blocks.  We didn't have time to pin-point the exact point of failure, so we edited /layout/mw_freegift.xml and commented out the code responsible for adding presentation to the product details page <catalog_product_view> (our template didn't take advantage of that presentation anyways.)  

This fixed our problem, and now simple products are reliably added to the cart.

I do like Phil's approach because it guarantees the desired add to cart behavior -- I will implement should I see this happen again.

Thanks for sharing,

Dan
